### DISCUSSION TITLE
How to customize error message when current statement is not allowed/ambiguous in grammar?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1694] [Q&A] [answered]
Hi Langium community, I am implementing validator for a DSL.
I am wondering if a statement is not complete in grammar such as statement without `;` which is shown below when our grammar requires `;` at the end of each statement, how to customize error message such as **"Expect ; at then end of statement"** in this case? 
![image](https://github.com/user-attachments/assets/8816176a-5ab4-4935-8591-0e2c1b32097f)

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

You can create a custom `ParserErrorMessageProvider`:

https://github.com/eclipse-langium/langium/blob/335f9a99b1568829cd1c0f5d3bbe33ca754daf90/packages/langium/src/services.ts#L62
